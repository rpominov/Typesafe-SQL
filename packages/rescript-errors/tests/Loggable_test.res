open Jest

test("fromText", () => {
  let loggable = Loggable.fromText("test")
  expect(loggable->Loggable.cause)->toEqual(None)
  expect(loggable->Loggable.compile)->toMatchSnapshot
  expect(loggable->Loggable.toString)->toEqual("test")
})

test("fromExn Invalid_argument", () => {
  let (loggable, exn) = try {
    raise(Invalid_argument("test"))
  } catch {
  | exn => (Loggable.fromExn(exn), exn)
  }
  expect(loggable->Loggable.cause)->toEqual(Exn(exn))
  expect(loggable->Loggable.compile)->toMatchSnapshot
  loggable
  ->Loggable.toString
  ->Js.String2.replaceByRe(%re("/at .+/g"), "at ...")
  ->expect
  ->toMatchSnapshot
})

test("fromExn Js.Exn.raiseError", () => {
  let loggable = try {
    Js.Exn.raiseError("test")
  } catch {
  | exn => Loggable.fromExn(exn)
  }
  expect(loggable->Loggable.cause)->toEqual(Native(Native.make("test")))
  expect(loggable->Loggable.compile)->toMatchSnapshot
  expect(loggable->Loggable.toString)->toEqual("test")
})

test("fromExnVerbose Js.Exn.raiseError", () => {
  let loggable = try {
    Js.Exn.raiseError("test")
  } catch {
  | exn => Loggable.fromExnVerbose(exn)
  }
  expect(loggable->Loggable.cause)->toEqual(Native(Native.make("test")))
  expect(loggable->Loggable.compile)->toMatchSnapshot
  expect((loggable->Loggable.toString->Js.String2.split("\n"))[0])->toEqual("Error: test")
})

test("fromExn Js.Exn.raiseError + prepend", () => {
  let loggable = try {
    Js.Exn.raiseError("test")
  } catch {
  | exn => Loggable.fromExn(exn)->Loggable.prepend("annotaion")
  }
  expect(loggable->Loggable.cause)->toEqual(Native(Native.make("test")))
  expect(loggable->Loggable.compile)->toMatchSnapshot
  expect(loggable->Loggable.toString)->toEqual("annotaion test")
})

test("fromExn Js.Exn.raiseError + append", () => {
  let loggable = try {
    Js.Exn.raiseError("test")
  } catch {
  | exn => Loggable.fromExn(exn)->Loggable.append("annotaion")
  }
  expect(loggable->Loggable.cause)->toEqual(Native(Native.make("test")))
  expect(loggable->Loggable.compile)->toMatchSnapshot
  expect(loggable->Loggable.toString)->toEqual("test annotaion")
})

test("fromExn (throw \"test\")", () => {
  let fn = %raw(`() => {throw "test"}`)

  let loggable = try {
    fn()
  } catch {
  | exn => Loggable.fromExn(exn)
  }
  expect(loggable->Loggable.cause)->toMatchSnapshot
  expect(loggable->Loggable.compile)->toMatchSnapshot
  expect(loggable->Loggable.toString)->toEqual(`'test'`)
})
