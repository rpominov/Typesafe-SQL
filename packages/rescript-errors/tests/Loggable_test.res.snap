// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fromExn (throw "test") 1`] = `
Object {
  "TAG": 2,
  "_0": "test",
}
`;

exports[`fromExn (throw "test") 2`] = `
Array [
  "test",
]
`;

exports[`fromExn Invalid_argument 1`] = `
Array [
  Object {
    "Error": [Error],
    "RE_EXN_ID": "Invalid_argument",
    "_1": "test",
  },
]
`;

exports[`fromExn Js.Exn.raiseError + append 1`] = `
Array [
  "test",
  "annotaion",
]
`;

exports[`fromExn Js.Exn.raiseError + prepend 1`] = `
Array [
  "annotaion",
  "test",
]
`;

exports[`fromExn Js.Exn.raiseError 1`] = `
Array [
  "test",
]
`;

exports[`fromExnVerbose Js.Exn.raiseError 1`] = `
Array [
  [Error: test],
]
`;

exports[`fromText 1`] = `
Array [
  "test",
]
`;
